tests/test_reader.py

```diff
@@ -3,35 +3,46 @@
 """
 
 import os
+import pytest
 import xml.etree.ElementTree as ET
 from glob import glob
 
 from pynxtools.dataconverter.helpers import (
     generate_template_from_nxdl,
     validate_data_dict,
 )
+from pynxtools.dataconverter.writer import Writer
 from pynxtools.dataconverter.template import Template
 from pynxtools.nexus.nxdl_utils import get_nexus_definitions_path
 
 from pynxtools_stm.reader import STMReader
 
 
-def test_example_data():
-    """
-    Test the example data for the stm reader
-    """
-    reader = STMReader
-    assert callable(reader.read)
+@pytest.mark.parametrize(
+    "nxdl,example_data",
+    [
+        ("NXsts", "data/sts_nanonis_5e"),
```

Comment:
Why limit yourself to NXsts and not just test against all nxdls in `reader.supported_nxdls´ (as it was before)? If there will be specializing appdefs, you could then also test against them.

Reply:
Or does not every data set work with all supported_nxdls?

Reply:
I will modify the test so that it also checks whether the `NXsts` already exist in `supported_nxdls` or not. 

Yes, every data set supports the existing `supported_nxdl` which is `NXsts`.

Reply:
Why parametrize the nxdl at all then? You could just go with the old approach of 
```
for supported_nxdl in reader.supported_nxdls:
    nxdl_file = os.path.join(
        def_dir, "contributed_definitions", f"{supported_nxdl}.nxdl.xml"
    )
    ...
```

Reply:
For it seems problematic for me. If I consider a specific sub-technique from a main technique, and a specific `nxdl` for that sub-tecnique, the this for loop does not work. Because, the data-set is not compatible with the general `nxdl` biuld for the main technique.

Reply:
Ok, so this is for future maintenance then, makes sense.